lib/components/private/InlineField/InlineField.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { ReactNode, AllHTMLAttributes } from 'react';
+import { useClassNames } from 'sku/treat';
+import * as styles from './InlineField.treat';
+import { Box } from '../../Box/Box';
+import { FieldLabelProps } from '../../FieldLabel/FieldLabel';
+import {
+  FieldMessage,
+  FieldMessageProps,
+} from '../../FieldMessage/FieldMessage';
+import { FieldOverlay } from '../FieldOverlay/FieldOverlay';
+import { Text } from '../../Text/Text';
+import { TickIcon } from '../../icons/TickIcon/TickIcon';
+
+type FormElementProps = AllHTMLAttributes<HTMLFormElement>;
+export interface InlineFieldProps {
+  id: NonNullable<FormElementProps['id']>;
+  label: NonNullable<FieldLabelProps['label']>;
+  onChange: NonNullable<FormElementProps['onChange']>;
+  checked: NonNullable<FormElementProps['checked']>;
+  name?: FormElementProps['name'];
+  disabled?: FormElementProps['disabled'];
+  message?: FieldMessageProps['message'];
+  tone?: FieldMessageProps['tone'];
+  children?: ReactNode;
+}
+
+interface InternalInlineFieldProps extends InlineFieldProps {
+  type: 'checkbox' | 'radio';
+}
+
+export const InlineField = ({
+  id,
+  name,
+  checked,
+  onChange,
+  label,
+  type,
+  children,
+  message,
+  tone = 'neutral',
+  disabled = false,
+}: InternalInlineFieldProps) => {
+  const messageId = `${id}-message`;
+  const isCheckbox = type === 'checkbox';
+  const radioStyles = {
+    [styles.circle]: type === 'radio',
+  };
+
+  return (
+    <Box>
```

Comment:
Can't use a `Fragment` here?

Reply:
Flipped on this a bit, but decided to keep one containing element for the field to keep it's children and layout more deterministic.